src/biogeochem/test/DustEmis_test/test_DustEmisZender2003.pf

```diff
@@ -0,0 +1,483 @@
+module test_DustEmisZender2003
+
+  ! Tests of DustEmisZender2003
+
+  use funit
+  use unittestSubgridMod
+  use unittestSimpleSubgridSetupsMod, only : setup_single_veg_patch
+  use clm_varpar, only : nlevsoi, nlevgrnd, nlevsno, clm_varpar_init, ndst
+  use clm_varctl, only : soil_layerstruct_predefined, create_crop_landunit, use_crop, create_crop_landunit
+  use clm_varcon, only : clm_varcon_init, clm_varcon_clean
+  use SnowHydrologyMod, only : InitSnowLayers, SnowHydrologySetControlForTesting, SnowHydrologyClean
+  use DustEmisBase
+  use DustEmisZender2003
+  use shr_kind_mod , only : r8 => shr_kind_r8
+  use unittestFilterBuilderMod, only : filter_from_range
+  use atm2lndType, only : atm2lnd_type, atm2lnd_params_type
+  use SoilStateType, only : soilstate_type
+  use CanopyStateType, only : canopystate_type
+  use TemperatureType, only : temperature_type
+  use WaterType, only : water_type
+  use FrictionVelocityMod, only : frictionvel_type
+  use unittestWaterTypeFactory, only : unittest_water_type_factory_type
+  use SoilStateInitTimeConstMod, only : ThresholdSoilMoistZender2003, ThresholdSoilMoistKok2014, MassFracClay
+  use DustEmisFactory, only : create_dust_emissions
+
+  implicit none
+
+  real(r8), parameter :: tol = 1.e-18_r8
+
+  @TestCase
+  type, extends(TestCase) :: TestDustEmisZender2003
+     class(dust_emis_base_type), allocatable :: dust_emis
+     integer, allocatable :: filter_nolakep(:)      ! non-lake filter (patches)
+     integer :: num_nolakep              ! number of patches in non-lake filter
+     type(atm2lnd_type) :: atm2lnd_inst
+     type(soilstate_type) :: soilstate_inst
+     type(canopystate_type) :: canopystate_inst
+     type(temperature_type) :: temperature_inst
+     type(unittest_water_type_factory_type) :: water_factory
+     type(water_type) :: water_inst
+     type(frictionvel_type) :: frictionvel_inst
+   contains
+     procedure :: setUp
+     procedure :: tearDown
+     procedure :: setupEnvironment
+     procedure :: create_atm2lnd
+     procedure :: create_fv
+     procedure :: print_values
+     procedure :: validate_patch
+  end type TestDustEmisZender2003
+
+contains
+
+  !-----------------------------------------------------------------------
+
+  subroutine setUp(this)
+    use ColumnType, only : col
+    class(TestDustEmisZender2003), intent(inout) :: this
+
+    character(len=5) :: NLFilename = 'none'
+    real(r8), allocatable :: snow_depth_col(:)
+    real(r8), allocatable :: urb_em(:)
+    integer :: begl, endl, begc, endc
+    integer :: c
+    type(atm2lnd_params_type) :: atm2lnd_params
+
+    soil_layerstruct_predefined = '20SL_8.5m'
```

Comment:
A suggestion that Adrianna had here was that maybe this could become something like setUpForTesting for InitForTesting that's part of the dustEmisBase class. That removes all the complexity here, and makes the unit test easier to follow. A good goal for unit tests is to make them dead simple to follow. But, this setupForTesting subroutine could also be used for functional testing (or for Bill's self-test type testing). So it allows wider possibilities.

I'm going to think about this more, but this might be a really good way to handle this and addresses the conversation above about putting this in a unit-testing shared place.

Reply:
I thought about this and went ahead and put the shared piece into the unit_test_shr directory for now. If functional tests are added later it would be good to refactor when they come in rather than now.